src/context.cpp

```diff
@@ -596,61 +699,172 @@ namespace librealsense
             if ( profiles.size() > 0 )
             {
                 _dev->open( realdds_profiles );
+
             }
 
             software_sensor::open( profiles );
         }
 
+        void handle_video_data( realdds::topics::device::image && dds_frame,
+                                std::shared_ptr< stream_profile_interface > prof, std::string name )
+        {
+            rs2_software_video_frame rs2_frame = {};
+
+            // prof parameter holds the real data, rs2_software_video_frame forces us to hold a pointer to it.
+            // Because we use syncer, not calling on_video_frame in the lifetime of this function, we need a shared_ptr
+            // that the syncer will hold till using the frame.
+            std::shared_ptr< rs2_stream_profile > prof_holder = std::make_shared< rs2_stream_profile >();
+            prof_holder->profile = prof.get();
+            rs2_frame.profile = prof_holder.get();
+
+            //Copying from dds into LibRS space, same as copy from USB backend.
+            //TODO - use memory pool or some other frame allocator
+            rs2_frame.pixels = new uint8_t[dds_frame.size];
```

Comment:
How about a `new std::vector( std::move( dds_frame.raw_data ))`?

Reply:
I.e.,
```
auto p_pixels_vector = new std::vector< uint8_t >( std::move( dds_frame.raw_data ));
rs2_frame.pixels = p_pixels_vector->data();
rs2_frame.deleter = [p_pixels_vector]( void * ) { delete p_pixels_vector; };

Reply:
The deleter is a plain C function `void(*deleter)(void*)` so I can't bind lambda parameters

Reply:
Added to the [to-do list](https://rsjira.intel.com/browse/LRS-585) a comment about changing the `deleter` API to receive a parameter.